# P01 - QIIME 2 Demultiplexing, Denoising and Clustering 

Feb 14th, 2025

## Purpose:
To import and sort the 16S rRNA sequences based on barcode information in QIIME2. To detect and correct sequencing errors, and to group the sequences into respective ASVs

## Material: 
1. QIIME2
2. manifest.tsv 
3. demux_seqs.qza
4. metadata.tsv 

## Method:
1. Connect to Cisco Anyconnect Secure VPN. Open the terminal and login to MICB 475 class server using the provided login credentials.
2. Create a dedicated directory for project 2: /data/team2
3. Import and demultiplex the dysautonomia dataset using pair-end to create a qza file with demultiplexed samples 
5. Create a visualization for the demultiplexed samples
6. Move the demux.qzv file to your local computer directory and view it using the QIIME 2
7. Create a detached screen and name it "denoising".
8. Denoise and cluster the demultiplexed sequences using DADA2
9. Visualize the ASVs by converting qza files to qzv.
10. Transfer the visualization files to local computer and view the representative sequences and table.qzv using view.QIIME2.org

## Code: 
[QIIME2 Code](QIIME2/processing_code.txt)

## Output files:
1. /pj2_export/demux_seqs.qzv
3. /data/team6_dysautonomia_data/team6_dysautonomia_table.qza 
2. /data/team6_dysautonomia_data/team6_dysautonomia_table.qzv - [file](/QIIME2/Data_processing/export/table.qzv)
3. /data/team6_dysautonomia_data/team6_rep-seqs.qza
4. /pj2_export/rep-seqs.qzv
5. /data/team6_dysautonomia_data/denoising-stats.qza
   
## Results: 
The same result is generated from samples sequenced from both forward and reverse primers
1. Total number of reads: 6531021 
2. Total number of samples: 214
3. Range of sequencing depth: 7440-114131
4. Maximum read length (bp): 251
5. All the reads the same length of 251 bp

## Results: AFTER DENOISING/CLUSTERING:
1. Total number of reads retained: 3,134,413  
2. Total number of ASVs: 2,463
3. Total number of samples: 214
4. Range of sequencing depth: 3,164 - 60,202

Sequence counts
> <img src="/QIIME2/Figure/Sequence Counts" height="200">

Sequence lengths
> <img src="/QIIME2/Figure/Sequence Length" height="300">

Attached images are generated from [https://view.qiime2.org/](https://view.qiime2.org/)

Table summary and frequency per sample 
> <img src="/QIIME2/Figure/Table summary and frequency per sample.png" height="400">

Sequence length statistics
> <img src="/QIIME2/Data_processing/figures/sequence_length_statistics.png" height="200">

Attached images are generated from [https://view.qiime2.org/](https://view.qiime2.org/)

## Discussion:
1. The maximum read depth (bp) was 251 while all 214 samples (the same number of samples using both forward and reverse primers) had 251 bp in length.
2. After demultiplexing, the medians of the quality score for all bases were consistently high with a similar median value of 30, presenting over 99.9% base call accuracy. Hence, this suggests that there would be no trimming required. The truncation length selected was 251 bp.
3. Samples sequenced using forward primers resulted in higher phred scores compared to that of the reverse primers, hence denoising step will use those amplified using forward primers
4. The total number of samples did not change since all the samples had reads of 150 bp in length and every read with 150 bp was retained.

## Future direction:
1. Denoise sequences using the selected truncation length of 251 bp and determine ASVs with DADA2 (outputs: team6_dysautonomia_table.qza, denoising-stats.qza)
2. The clustered ASVs can be filtered to exclude any mitochondrial or chloroplast sequences, which will remove any eukaryotic data.
