# P02 - QIIME2 Taxonomic Analysis and Data Export for R use

Feb 18, 2024

## Purpose:
To train a classifier for taxanomic analysis and export all data in QIIME2 to a format suitable for R processing. 

## Material: 
1. QIIME2
2. /mnt/datasets/silva_ref_files/silva-138-99-seqs.qza
3. /mnt/datasets/silva_ref_files/silva-138-99-tax.qza
4. /data/team_2/rep-seqs.qza

## Method:
1. Extract the amplicon of interest from the reference database
2. Train classifier with your new ref-seq file

## Code: 
[Taxonomic Analysis](/QIIME2/P01_Code.txt)

## Output files:
1. /data/team_2/ref-seqs-trimmed.qza
2. /data/team_2/classifier.qza
3. /data/team_2/taxonomy.qza
4. /data/team_2/aligned-rep-seqs.qza
5. /data/team_2/masked-aligned-rep-seqs.qza
6. /data/team_2/unrooted-tree.qza
7. /data/team_2/rooted-tree.qza

## Results: 
Rarefaction curve:
![Rarefaction Curve](/QIIME2/Figure/Rarefaction_uncropped.png)

Rarefaction curve with the selected sampling depth indicated by the black vertical line:
![Rarefaction Curve](/QIIME2/Figure/Rarefaction_cropped.png)

## Discussion:
To determine the rarefaction depth, we performed rarecurve() on R. As shown in Figure 2, the majority of species plateaued at a sequencing depth of 1500. However, after reviewing the samples retained across all metadata categories at a rarefaction depth of 1500, we found that we lost all pre-ICI stool and spleen samples. Therefore, to ensure a minimum of 3 samples per organ site (excluding stool), we selected a rarefaction depth of 650. Unfortunately, stool samples contained very few reads from the start, as a result, we were only able to retain 1 pre-ICI stool sample after rarefaction. The sampling depth retained 1838 (73.26%) features in 184 (67.90%) samples at the specified sample depth. Here we showed both the original rarefaction plot, and the zoomed-in plot, where the blue line indicates the location of the sampling depth.

## Future direction:
We will proceed with conducting taxonomic analysis for each group.
