# P02 - QIIME2 Taxonomic Analysis and Data Export for R use

Feb 18, 2024

## Purpose:
To export all data in QIIME2 to a format suitable for R processing. 

## Material: 
1. QIIME2
2. R & Rstudio
3. 

## Method:
1. Format OTU table, sample metadata, and taxonomy
2. Create phyloseq object by merging formatted OTU table, sample metadata, taxonomy, and phylogeny tree
3. Filter to remove ASVs that have less than 5 counts total and samples with less than 100 reads
4. Rarify to a sampling depth of 9069.

## Code: 
[Phyloseq and Rarecurve code](/R_Project/FD_phyloseq_&_Rarecurve.R)

## Output files:
1. FD_final.Rdata (Phyloseq object)
2. FD_rare.Rdata (Rareified sample data)

## Results: 
Rarefaction curve with the selected sampling depth indicated by the black vertical line
![Rarefaction Curve](https://github.com/oliviakwon/MICB475_Team6/assets/158798155/89706f6e-3fd0-404f-89c0-542f6405eb77)

## Discussion:
Based on our rarefaction curve, above ~6000 seems a good sampling depth since most of the samples have already saturated the representation of ASVs. 
A sampling depth of 9069 was chosen because, at this sampling depth, we can retain 1,414,764 (45.30%) features and 156(72.90%) samples.

## Future direction:
Identify which variables are significantly associated with gut microbiome dysbiosis via a univariate regression analysis.
