# P02 - QIIME2 Taxonomic Analysis and Data Export for R use

Feb 18, 2024

## Purpose:
To train a classifier for taxanomic analysis and export all data in QIIME2 to a format suitable for R processing. 

## Material: 
1. QIIME2
2. /mnt/datasets/silva_ref_files/silva-138-99-seqs.qza
3. /mnt/datasets/silva_ref_files/silva-138-99-tax.qza
4. /data/team_2/rep-seqs.qza

## Method:
1. Extract the amplicon of interest from the reference database
2. Train classifier with your new ref-seq file

## Code: 
[Taxonomic Analysis](/QIIME2/P01_Code.txt)

## Output files:
1. /data/team_2/ref-seqs-trimmed.qza
2. /data/team_2/classifier.qza
3. /data/team_2/taxonomy.qza
4. /data/team_2/aligned-rep-seqs.qza
5. /data/team_2/masked-aligned-rep-seqs.qza
6. /data/team_2/unrooted-tree.qza
7. /data/team_2/rooted-tree.qza

## Results: 
Rarefaction curve with the selected sampling depth indicated by the black vertical line
![Rarefaction Curve](/QIIME2/Figure/Rarefaction_original.png)

## Discussion:
Based on our rarefaction curve, above ~6000 seems a good sampling depth since most of the samples have already saturated the representation of ASVs. 
A sampling depth of 9069 was chosen because, at this sampling depth, we can retain 1,414,764 (45.30%) features and 156(72.90%) samples.

## Future direction:
Identify which variables are significantly associated with gut microbiome dysbiosis via a univariate regression analysis.
